### ISSUE TITLE
Outlet not working[Bug]: 

### ISSUE THREAD

**Issue by sirEmmyUche** [remix-run/react-router#10764]
### What version of React Router are you using?

v 6.14.2

### Steps to Reproduce
```js
import {Outlet} from "react-router-dom"
import Nav from "../pages/nav.jsx"

function Layout(){
    <div>
        <Nav/>
        <main>
            <Outlet/>
        </main>
    </div>
}

export default Layout

import {RouterProvider,
  createBrowserRouter,
  Routes,
  Route,
} from "react-router-dom"
import Home from "./pages/home"
import About from "./pages/about"
import Layout from "./layouts/layout"

const router = createBrowserRouter([
  {path:"*", Component:Root}
])

function Root(){
  return (
    <Routes>
        <Route element={<Layout/>}>
          <Route path="/" element={<Home/>}/>
          <Route path="/about" element={<About/>}/>
        </Route>
    </Routes>
  )
}

function App(){
  return < RouterProvider router={router}/>
}

export default App
```

### Expected Behavior

When Outlet is being used, It is supposed to display the children route that are wrapped within the pathless route. 

for example:
```js
import {Outlet} from "react-router-dom"
import Nav from "../pages/nav.jsx"

function Layout(){
    <div>
        <Nav/>
        <main>
            <Outlet/>
        </main>
    </div>
}

export default Layout
```
Now In the app component
```js
 <Routes>
        <Route element={<Layout/>}>
          <Route path="/" element={<Home/>}/>
          <Route path="/about" element={<About/>}/>
        </Route>
    </Routes>
```
It is expected that the home and about page should be displayed having a fixed nav on both page.

### Actual Behavior

Everything goes blank. I also checked some project that I have worked on before which was working very fine however, this time everything goes blank in  despite they have been deployed to production and was working fine previously. 

after much debugging, I found out that whenever  the Outlet is removed, it works fine.

**Comment by sirEmmyUche** [comment]
Outlet needs some fix.

**Comment by timdorr** [comment]
You are mixing a data router with the older component-based routes. They are not compatible. You need to choose one or other other for Outlet to work.

**Comment by sirEmmyUche** [comment]
Hello Remix-Run/React-Router,

I saw your reply as regarding the issue I submitted (issue #10764 ~Outlet
not working)

I still don't get the "mix data router" you're referring to. Moreover, I
want to point the fact that other project on production has currently
crashed. Mind you, these projects were working on a different version, they
were not upgraded to the newer versions and they were working fine as at
the time of deployment. But right now, when the Outlet router is used, it
crashes them.

Please kindly provide more clarity on the "mix data router" you're talking
about.

Regards.

**Comment by kiliman** [comment]
He means you're using `createBrowserRouter`, but using `Routes` components instead of passing in the routes config.

You should either do this:
```ts
<BrowserRouter>
  <Routes>
    ...
  </Routes>
</BrowserRouter>
```
or
```ts
const router = createBrowserRouter([
  {
    element: <Layout />,
    children: [
      {
        path: "/",
        element: <Home />,
      },
      {
        path: "/about",
        element: <About />,
      },
    ],
  },
]);
```
Or you can use the helper `createRoutesFromElements`
```ts
const router = createBrowserRouter(createRoutesFromElements(
   <Routes>
    ...
  </Routes>
)
```

https://reactrouter.com/en/main/utils/create-routes-from-elements

**Comment by brophdawg11** [comment]
There's a bit of confusion here.  

You _can't_ mix `<BrowserRouter>`/`<RouterProvider>`, but you _can_ nest a `<Routes>` sub-tree inside a `RouterProvider` splat route (as this example is doing) - this is useful if you are [migrating from `<Routes>` to `createBrowserRouter` route-by-route](https://reactrouter.com/en/main/upgrading/v6-data).

I'm not sure what your issue is but this setup should work fine, as show in this code sandbox: https://codesandbox.io/s/wonderful-tdd-wqt88y

If you can provide a reproduction of this approach not working, please-re-open this issue with a codesandbox or github repo and I can dig in deeper 👍

**Comment by sirEmmyUche** [comment]
Yeah! Thanks, I've seen the error. It's been fixed.

Thanks

**Comment by Lokesh396** [comment]
This is my App.jsx
`import './styles.css'
import { BrowserRouter, Routes, Route } from 'react-router-dom'
import Home from './components/Home'
import About from './components/pages/About'
import Resume from './components/pages/Resume'
import Work from './components/pages/Work'
import Contact from './components/pages/Contact'
function App() {
  return (
    <>
    <BrowserRouter>

        <Routes>
          <Route path='/' element={<Home/>}>
            <Route path='about' element={About}></Route>
            <Route path='resume' element={Resume}></Route>
            <Route path='work' element={Work}></Route>
            <Route path='contact' element={Contact}></Route>
          </Route>
        </Routes>

    </BrowserRouter>
     
    </>
  )
}

export default App
`
This is Home.jsx

`import NavBar from "./pages/NavBar";
import Header from "./pages/Header";
import { Outlet } from "react-router-dom";
import Profile from "./pages/Profile";
function Home() {
  return (
    <>
      <Header></Header>
      <div className="d-flex gap-3">
          <Profile></Profile>
        <div className="d-felx gap-3">
        <NavBar></NavBar>
          <Outlet/>
        </div>
      </div>
    </>
  );
}

export default Home;
`
I am using react: 18.2.0
react-router-dom: 6.17.0

If I  navigate to http://localhost:5173/about, the about page is not rendering, but i can see header, profile, and navbar

**Comment by diana96alazzam** [comment]
I also have some kind of inconsistency in `Outlet` behavior and I'm not sure what should I do to make it work.

I have a custom component in my shared components package that returns a `RouterProvider`. I also have another layout component that is rendering a menu with an `Outlet` and uses "`useLocation`" hook. 

- If I use these two components from the package in another app, the outlet doesn't work and it doesn't render the component that should be rendered in its place. 

- If I use `RouterProvider` directly in the application and pass routes to it, I get an error saying that I cannot use "`useLocation`" outside of a Router. This happens because the element that is rendered contains the component that has `useLocation` hook.

This issue only happens when using the components exported from the package, because I tried passing the same router array to a `RouterProvider` directly and it worked fine, but when I passed the same router array to the component exported from my package, it only rendered the first component and ignored the outlet. 

Any idea on what is happening and what can I do to make the outlet work in another projects that are using this shared components package?

Thanks
